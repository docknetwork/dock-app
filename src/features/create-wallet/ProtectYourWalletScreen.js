import React, {useEffect} from 'react';
import {
  Button,
  Header,
  Footer,
  Content,
  ScreenContainer,
  Typography,
  Box,
  BigButton,
  runAfterInteractions,
} from '../../design-system';
import {BackButton} from '../../design-system/buttons';
import EmojiHappyIcon from '../../assets/icons/emoji-happy.svg';
import ArrowRight from '../../assets/icons/arrow-right.svg';
import FingerprintSimple from '../../assets/icons/fingerprint-simple.svg';
import {useDispatch, useSelector} from 'react-redux';
import {walletOperations} from './wallet-slice';
import {appSelectors, BiometryType} from '../app/app-slice';

// const BigButton = ({icon, children, ...props}) => (
//   <Box
//     row
//     borderWidth={1}
//     borderColor="#3F3F46"
//     borderRadius={8}
//     padding={25}
//     marginBottom={12}
//     alignItems="center"
//     {...props}>
//     <Box autoSize col>
//       {icon}
//     </Box>
//     <Box>
//       <Typography
//         fontSize={14}
//         fontFamily="Nunito Sans"
//         fontWeight="600"
//         color="#fff">
//         {children}
//       </Typography>
//     </Box>
//   </Box>
// );

export function ProtectYourWalletScreen({
  onSkip,
  onEnable,
  faceId,
  fingerprint,
}) {
  return (
    <ScreenContainer testID="create-wallet-screen">
      <Header>
        <BackButton />
      </Header>
      <Content marginLeft={26} marginRight={26}>
        <Typography
          fontFamily="Montserrat"
          fontSize={24}
          lineHeight={32}
          fontWeight="600"
          color="#fff"
          marginTop={52}>
          Protect your wallet
        </Typography>
        <Typography
          fontSize={16}
          lineHeight={24}
          fontWeight="400"
          marginTop={12}>
          Use your biometric to add extra layer of security and easier way to
          access your wallet
        </Typography>
      </Content>
      <Footer marginBottom={114} marginLeft={26} marginRight={26} flex>
        {/* <Button full icon={<LockClosedIcon />} testID="create-wallet-btn">
          Setup Passcode
        </Button> */}
        {faceId ? (
          <BigButton onPress={onEnable} icon={<EmojiHappyIcon />}>
            Use Face recognition
          </BigButton>
        ) : null}
        {fingerprint ? (
          <BigButton onPress={onEnable} icon={<FingerprintSimple />}>
            Use Fingerprint identification
          </BigButton>
        ) : null}
        <BigButton onPress={onSkip} icon={<ArrowRight />}>
          Do this later
        </BigButton>
      </Footer>
    </ScreenContainer>
  );
}

export function ProtectYourWalletContainer() {
  const dispatch = useDispatch();
  const supportedBiometryType = useSelector(
    appSelectors.getSupportedBiometryType,
  );
  // detect biometry
  let fingerprint = supportedBiometryType === BiometryType.Fingerprint;
  let faceId = supportedBiometryType === BiometryType.FaceId;

  const handleEnable = () => {
    runAfterInteractions(() => dispatch(walletOperations.createWallet({biometry: true})));
  };

  const handleSkip = () => {
    runAfterInteractions(() => dispatch(walletOperations.createWallet()));
  };

  useEffect(() => {
    if (!faceId && !fingerprint) {
      handleSkip();
    }
  }, [faceId, fingerprint]);

  return (
    <ProtectYourWalletScreen
      fingerprint={fingerprint}
      faceId={faceId}
      onEnable={handleEnable}
      onSkip={handleSkip}
    />
  );
}
