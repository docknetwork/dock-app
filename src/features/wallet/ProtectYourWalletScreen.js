import React, {useCallback, useEffect, useState} from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {ScreenSpinner} from 'src/components/ScreenSpinner';
import {showToast} from 'src/core/toast';
import ArrowRight from '../../assets/icons/arrow-right.svg';
import EmojiHappyIcon from '../../assets/icons/emoji-happy.svg';
import FingerprintSimple from '../../assets/icons/fingerprint-simple.svg';
import {
  BigButton,
  Content,
  Footer,
  Header,
  runAfterInteractions,
  ScreenContainer,
  Typography,
} from '../../design-system';
import {BackButton} from '../../design-system/buttons';
import {appSelectors, BiometryType} from '../app/app-slice';
import {WalletConstants} from './constants';
import {walletOperations} from './wallet-slice';

export function ProtectYourWalletScreen({
  onSkip,
  onEnable,
  faceId,
  fingerprint,
}) {
  return (
    <ScreenContainer testID="create-wallet-screen">
      <Header>
        <BackButton />
      </Header>
      <Content marginLeft={26} marginRight={26}>
        <Typography variant="h1" marginTop={52}>
          Protect your wallet
        </Typography>
        <Typography marginTop={12}>
          Use your biometric to add extra layer of security and easier way to
          access your wallet
        </Typography>
      </Content>
      <Footer marginBottom={114} marginLeft={26} marginRight={26} flex>
        {/* <Button full icon={<LockClosedIcon />} testID="create-wallet-btn">
          Setup Passcode
        </Button> */}
        {faceId ? (
          <BigButton onPress={onEnable} icon={<EmojiHappyIcon />}>
            Use Face recognition
          </BigButton>
        ) : null}
        {fingerprint ? (
          <BigButton onPress={onEnable} icon={<FingerprintSimple />}>
            Use Fingerprint identification
          </BigButton>
        ) : null}
        <BigButton onPress={onSkip} icon={<ArrowRight />}>
          Do this later
        </BigButton>
      </Footer>
    </ScreenContainer>
  );
}

export function ProtectYourWalletContainer() {
  const dispatch = useDispatch();
  const [loading, setLoading] = useState(false);
  const supportedBiometryType = useSelector(
    appSelectors.getSupportedBiometryType,
  );
  // detect biometry
  let fingerprint = supportedBiometryType === BiometryType.Fingerprint;
  let faceId = supportedBiometryType === BiometryType.FaceId;

  const handleEnable = () => {
    setLoading(true);
    runAfterInteractions(() => {
      dispatch(walletOperations.createWallet({biometry: true}))
        .then(() => {
          showToast({
            message: WalletConstants.protectYourWallet.locales.biometicsEnabled,
            type: 'success',
          });
        })
        .catch(() => {
          showToast({
            message: WalletConstants.protectYourWallet.locales.biometicsFailed,
            type: 'error',
          });
        })
        .finally(() => setLoading(false));
    });
  };

  const handleSkip = useCallback(() => {
    runAfterInteractions(() => dispatch(walletOperations.createWallet()));
  }, [dispatch]);

  useEffect(() => {
    if (!faceId && !fingerprint) {
      handleSkip();
    }
  }, [faceId, fingerprint, handleSkip]);

  if (loading) {
    return <ScreenSpinner />;
  }

  return (
    <ProtectYourWalletScreen
      fingerprint={fingerprint}
      faceId={faceId}
      onEnable={handleEnable}
      onSkip={handleSkip}
    />
  );
}
