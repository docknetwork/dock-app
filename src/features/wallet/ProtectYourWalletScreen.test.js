import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {ProtectYourWalletScreen} from './ProtectYourWalletScreen';

const mockStore = configureMockStore();

describe('ProtectYourWalletScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(<ProtectYourWalletScreen />, {
      context: {store: mockStore(initialState)},
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
