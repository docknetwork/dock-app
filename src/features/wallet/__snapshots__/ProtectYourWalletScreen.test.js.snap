// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectYourWalletScreen should render correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#18181B",
      "flex": 1,
    }
  }
  testID="create-wallet-screen"
>
  <AppGlobalHeader />
  <Styled(View)>
    <BackButton />
  </Styled(View)>
  <Styled(ScrollView)
    marginLeft={26}
    marginRight={26}
  >
    <Styled(Component)
      marginTop={52}
      variant="h1"
    >
      Protect your wallet
    </Styled(Component)>
    <Styled(Component)
      marginTop={12}
    >
      Use your biometric to add extra layer of security and easier way to access your wallet
    </Styled(Component)>
  </Styled(ScrollView)>
  <Styled(Box)
    flex={1}
    marginBottom={114}
    marginLeft={26}
    marginRight={26}
  >
    <BigButton
      accessibilityLabel="DoThisLater"
      icon={<SvgFile />}
      testID="DoThisLater"
    >
      Do this later
    </BigButton>
  </Styled(Box)>
</RNCSafeAreaView>
`;
